client/me/purchases/manage-purchase/change-payment-method/index.jsx

```diff
@@ -216,35 +233,71 @@ function ChangePaymentMethodList() {
 					translate={ translate }
 					disabled={ disabled }
 					formSubmitting={ formSubmitting }
+					purchase={ purchase }
+					setFormSubmitting={ setFormSubmitting }
+					successCallback={ successCallback }
 				/>
 			</Card>
 		</CheckoutProvider>
 	);
 }
 
-function SaveButton( { translate, disabled, formSubmitting } ) {
-	const [ paymentMethodId ] = usePaymentMethodId();
+function SaveButton( {
+	translate,
+	disabled,
+	formSubmitting,
+	purchase,
+	setFormSubmitting,
+	successCallback,
+} ) {
+	const [ selectedPaymentMethodId ] = usePaymentMethodId();
+	const { showErrorMessage, showSuccessMessage } = useMessages();
 
-	// TODO: make sure the button busy state works correctly
 	const isSubmitting = disabled || formSubmitting;
 
+	const isStoredCard = /^existingCard-/.test( selectedPaymentMethodId );
+
 	const onClick = () => {
-		// TODO: do the actual submission here
-		window.alert( 'YOU CLICKED SUBMIT ' + paymentMethodId );
+		setFormSubmitting( true );
+		wpcomAssignPaymentMethod( purchase.id, selectedPaymentMethodId.replace( /^existingCard-/, '' ) )
+			.then( () => {
+				showSuccessMessage( translate( 'Your payment method has been set.' ) );
+				setFormSubmitting( false );
+				successCallback();
+			} )
+			.catch( ( error ) => {
+				showErrorMessage( error );
+				setFormSubmitting( false );
+			} );
 	};
 
+	let buttonText = null;
+	if ( isStoredCard ) {
+		buttonText = formSubmitting
+			? translate( 'Saving card…', {
+					context: 'Button label',
+					comment: 'Credit card',
+			  } )
+			: translate( 'Use this card', {
```

Comment:
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.